app/src/main/java/ch/epfl/favo/common/CollectionWrapper.java

```diff
@@ -1,22 +1,38 @@
 package ch.epfl.favo.common;
 
+import java.util.List;
 import java.util.Map;
+import java.util.concurrent.CompletableFuture;
 
-public class CollectionWrapper implements DatabaseUpdater {
+public class CollectionWrapper<T extends Document> implements DatabaseUpdater<T> {
 
-  private static final String TAG = "DatabaseWrapper";
-  private String collectionReference;
+  private String collection;
+  private Class cls;
 
-  public CollectionWrapper(String collectionReference) {
-    this.collectionReference = collectionReference;
+  public CollectionWrapper(String collection, Class cls) {
+    this.collection = collection;
+    this.cls = cls;
   }
 
-  public void addDocument(String key, Map document) {
-    DatabaseWrapper.addDocument(key, document, collectionReference);
+  @Override
+  public void addDocument(T document) {
+    DatabaseWrapper.addDocument(document, collection);
+  }
+
+  public void removeDocument(String key) {
+    DatabaseWrapper.removeDocument(key, collection);
+  }
+
+  public void updateDocument(String key, Map<String, Object> updates) {
+    DatabaseWrapper.updateDocument(key, updates, collection);
   }
 
   @Override
-  public Map<String, Object> getDocument(String key) {
-    return DatabaseWrapper.getDocument(key, collectionReference);
+  public CompletableFuture<T> getDocument(String key) {
+    return DatabaseWrapper.getDocument(key, cls, collection);
+  }
+
+  public CompletableFuture<List<T>> getAllDocuments() {
```

Comment:
Not sure if we need a collectionwrapper and a database wrapper. We can for example keep the collection as a field of the db wrapper and change it whenever we want to change collections

Reply:
We would be instantiating multiple instances of `DatabaseWrapper` in that case.